core/execution/future/market_snapshot.go

```diff
@@ -294,6 +301,16 @@ func (m *Market) GetState() *types.ExecMarket {
 	sort.Strings(parties)
 	assetQuantum, _ := m.collateral.GetAssetQuantum(m.settlementAsset)
 
+	partyMarginFactors := make([]*snapshot.PartyMarginFactor, 0, len(m.partyMarginFactor))
+	for k, d := range m.partyMarginFactor {
+		if !d.IsZero() {
```

Comment:
In what situation could this be 0?

Reply:
in some situation I forgot to remove it from the map.

Reply:
lol ok.

Reply:
This doesn't make sense though, forgetting to remove it from the map doesn't make the value become zero? 

Forgetting to remove it from the map will just leave someone in cross-margin mode with a non-zero margin factor, left as a non-zero margin factor that was set originally which is we get into that state will cause loops like this to go wonky:
```
for k, d := range m.partyMarginFactor {
		if !d.IsZero() {
			partyPos, _ := m.position.GetPositionByPartyID(k)
			if partyPos != nil && (partyPos.Buy() != 0 || partyPos.Sell() != 0) {
				marketObservable, mpos, increment, _, _, orders, err := m.getIsolatedMarginContext(partyPos, nil)
				if err != nil {
					continue
				}
				r := m.risk.ReleaseExcessMarginAfterAuctionUncrossing(ctx, mpos, marketObservable, increment, d, orders)
				if r != nil && r.Transfer() != nil {
					m.transferMargins(ctx, []events.Risk{r}, nil)
				}
			}
		}
	}
```

So I don't think the `if !d.IsZero()` guards are doing anything at all.

Reply:
no. that's not what I meant. what I meant is setting it to zero and still not removing it from the map. Hope it makes sense.

Reply:
but given that it seems to be impossible to set it to 0 in the map (considering it is only set when the margin mode passed to the switch is isolated margin, in which case the validation won't ket it get there with a margin factor of 0), then it seems safe to remove the checks.